File: Languages/C/math.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

// Hello, World!
void hello() {
    printf("Hello, World!\n");
}

// Simple expression evaluator
double evaluate_expression(const char* expr) {
    char* expr_copy = strdup(expr);
    char* token = strtok(expr_copy, " ");
    
    if (!token) return 0.0;
    
    double a = atof(token);
    
    token = strtok(NULL, " ");
    if (!token) {
        free(expr_copy);
        return a;
    }
    
    char op = token[0];
    
    token = strtok(NULL, " ");
    if (!token) {
        free(expr_copy);
        return a;
    }
    
    double b = atof(token);
    
    double result;
    switch (op) {
        case '+': result = a + b; break;
        case '-': result = a - b; break;
        case '*': result = a * b; break;
        case '/': result = b != 0 ? a / b : 0.0; break;
        default: result = 0.0; break;
    }
    
    free(expr_copy);
    return result;
}

void solve_equation(const char* line) {
    char equation[256];
    strncpy(equation, line, sizeof(equation) - 1);
    equation[sizeof(equation) - 1] = '\0';
    
    // Remove "= ?" part
    char* equals_pos = strstr(equation, "=");
    if (equals_pos) {
        *equals_pos = '\0';
    }
    
    // Trim whitespace
    char* end = equation + strlen(equation) - 1;
    while (end > equation && isspace((unsigned char)*end)) end--;
    *(end + 1) = '\0';
    
    double result = evaluate_expression(equation);
    printf("%s = %.2f\n", equation, result);
}

void process_file(const char* filename) {
    FILE* file = fopen(filename, "r");
    if (!file) {
        printf("Could not open file: %s\n", filename);
        return;
    }
    
    char line[256];
    while (fgets(line, sizeof(line), file)) {
        // Remove newline
        line[strcspn(line, "\n")] = 0;
        
        // Skip empty lines and markdown headers
        if (strlen(line) == 0 || line[0] == '#') continue;
        
        // Handle markdown list items
        char* start = line;
        if (strncmp(line, "- ", 2) == 0) {
            start = line + 2;
        }
        
        if (strstr(start, "=")) {
            solve_equation(start);
        }
    }
    
    fclose(file);
}

int main() {
    hello();
    
    printf("\nProcessing math equations...\n");
    
    process_file("../test_data/math_equations.txt");
    process_file("../test_data/math_equations.md");
    process_file("../test_data/math_equations.json");
    process_file("../test_data/math_equations.yaml");
    process_file("../test_data/math_equations");
    
    return 0;
}

File: Languages/Swift/math.swift
import Foundation

// Hello, World!
print("Hello, World!")

print("\nProcessing math equations...")

func evaluateExpression(_ expr: String) -> Double {
    let trimmedExpr = expr.trimmingCharacters(in: .whitespaces)
    
    // Use NSExpression to evaluate mathematical expressions
    let nsExpression = NSExpression(format: trimmedExpr)
    return nsExpression.expressionValue(with: nil, context: nil) as? Double ?? 0.0
}

func solveEquation(_ line: String) {
    var equation = line
    
    // Remove "= ?" part
    if let equalsRange = equation.range(of: "=") {
        equation = String(equation[..<equalsRange.lowerBound])
    }
    
    // Trim whitespace
    equation = equation.trimmingCharacters(in: .whitespaces)
    
    let result = evaluateExpression(equation)
    print("\(equation) = \(String(format: "%.2f", result))")
}

func processFile(_ filename: String) {
    guard let content = try? String(contentsOfFile: filename) else {
        print("Could not open file: \(filename)")
        return
    }
    
    let lines = content.components(separatedBy: .newlines)
    
    for line in lines {
        // Skip empty lines and markdown headers
        if line.isEmpty || line.hasPrefix("#") {
            continue
        }
        
        // Handle markdown list items
        let start: String
        if line.hasPrefix("- ") {
            start = String(line.dropFirst(2))
        } else {
            start = line
        }
        
        if start.contains("=") {
            solveEquation(start)
        }
    }
}

// Process all files
processFile("../test_data/math_equations.txt")
processFile("../test_data/math_equations.md")
processFile("../test_data/math_equations.json")
processFile("../test_data/math_equations.yaml")
processFile("../test_data/math_equations")
